### ISSUE TITLE
Can not install a package I created(stat: No such file or directory)

### ISSUE THREAD

**Issue by jiyinyiyong** [keplerproject/luarocks#402]
I created a repo https://github.com/Cirru/parser.moon

and I managed to run `luarocks make <.rockspec>` and `luarocks pack <.rockspec>` locally.

Then I uploaded `*.rockspec` to http://luarocks.org but failed to install

```
➤➤ luarocks install cirru-parser
Installing https://rocks.moonscript.org/cirru-parser-0.1-1.rockspec...
Using https://rocks.moonscript.org/cirru-parser-0.1-1.rockspec... switching to 'build' mode
Cloning into 'parser.moon'...
remote: Counting objects: 36, done.
remote: Compressing objects: 100% (26/26), done.
remote: Total 36 (delta 0), reused 35 (delta 0), pack-reused 0
Receiving objects: 100% (36/36), 8.07 KiB | 0 bytes/s, done.
Checking connectivity... done.
stat: cirru-parser/parser.lua: stat: No such file or directory

Error: Build error: Failed installing cirru-parser/parser.lua in /usr/local/lib/luarocks/rocks/cirru-parser/0.1-1/lua/cirru-parser.lua
```

That file is surely in the repo.

http://rocks.moonscript.org/cirru-parser-0.1-1.rockspec

```lua
package = 'cirru-parser'
version = '0.1-1'
source = {
  url = 'git://github.com/Cirru/parser.moon'
}
description = {
  summary = "Cirru Parser in Lua(MoonScript)",
  detailed = [[
    Mannual converted from CoffeeScript
  ]],
  homepage = 'https://github.com/Cirru/parser.moon',
  license = "MIT"
}
dependencies = {
  "lua >= 5.1"
}
build = {
  type = "module",
  modules = {
    ["cirru-parser"] = "cirru-parser/parser.lua",
    ["cirru-parser.tree"] = "cirru-parser/tree.lua",
    ["cirru-parser.array"] = "cirru-parser/array.lua"
  }
}
```

It's weird. But how is that supposed to work?

**Comment by Tieske** [comment]
Can you set `verbose=true` in your LuaRocks config file?

**Comment by jiyinyiyong** [comment]
I don't have a config file, hope this is the same:
```
➤➤ luarocks install cirru-parser --verbose

os.execute: 	unzip -n '/Users/chen/.cache/luarocks/https___rocks.moonscript.org/manifest-5.2.zip'
Archive:  /Users/chen/.cache/luarocks/https___rocks.moonscript.org/manifest-5.2.zip
  inflating: manifest-5.2
Results: 3
  1 (boolean): true
  2 (string): exit
  3 (number): 0
Installing https://rocks.moonscript.org/cirru-parser-0.1-1.rockspec...
Using https://rocks.moonscript.org/cirru-parser-0.1-1.rockspec... switching to 'build' mode

os.execute: 	'git' 'clone' '--depth=1' 'git://github.com/Cirru/parser.moon' 'parser.moon'
Cloning into 'parser.moon'...
remote: Counting objects: 36, done.
remote: Compressing objects: 100% (26/26), done.
remote: Total 36 (delta 0), reused 34 (delta 0), pack-reused 0
Receiving objects: 100% (36/36), 8.06 KiB | 0 bytes/s, done.
Checking connectivity... done.
Results: 3
  1 (boolean): true
  2 (string): exit
  3 (number): 0

io.popen: 	/usr/bin/stat -f '%A' 'cirru-parser/tree.lua'
stat: cirru-parser/tree.lua: stat: No such file or directory

Error: Build error: Failed installing cirru-parser/tree.lua in /usr/local/lib/luarocks/rocks/cirru-parser/0.1-1/lua/cirru-parser/tree.lua
```

**Comment by mpeterv** [comment]
This is a bug in `fetch.url_to_base_dir` which strips everything that looks like an extension from repo name.

**Comment by jiyinyiyong** [comment]
My project is running out of names so I chose to use extension. https://github.com/Cirru?utf8=%E2%9C%93&query=parser

**Comment by mpeterv** [comment]
As a workaround you can set `source.dir` in the rockspec explicitly, e.g. change

```lua
source = {
  url = 'git://github.com/Cirru/parser.moon'
}
```

to

```lua
source = {
  url = 'git://github.com/Cirru/parser.moon',
  dir = 'parser.moon'
}
```

**Comment by jiyinyiyong** [comment]
@mpeterv Thanks. It's working now.

I suppose if this is a wontfix bug, printing warnings about using `source.dir` would be helpful.

**Comment by siffiejoe** [comment]
There is also another open issue for this: #375.

**Comment by hishamhm** [comment]
> I suppose if this is a wontfix bug

It isn't, we will fix it :)

**Comment by ignacio** [comment]
fixed by 634ff0af63a250859863cee25b8ab21e2404a267
